src/seedu/addressbook/parser/Parser.java

```diff
@@ -69,6 +75,8 @@ public Command parseCommand(String userInput) {
             return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
         }
 
+        commandHistory.add(userInput);
```

Comment:
I'm not 100% sure but this adds the entire user input into ArrayList right? Doing this will add invalid inputs, etc, that may not be very necessary. Is this intended? Otherwise you might want to add an if condition to check validity of the command before adding it into ArrayList (matches())

Reply:
Yup, this adds all commands entered thus far including the invalid inputs just like the history command in Linux. However, I think your suggestion is good as well and would be more useful in this application context.